src/pybind/mgr/dashboard/controllers/host.py

```diff
@@ -145,3 +173,51 @@ def daemons(self, hostname: str) -> List[dict]:
         orch = OrchClient.instance()
         daemons = orch.services.list_daemons(None, hostname)
         return [d.to_json() for d in daemons]
+
+    @Endpoint('GET')
+    @ReadPermission
+    @handle_orchestrator_error('host')
+    def labels(self) -> List[str]:
+        """
+        Get all host labels.
+        Note, host labels are only supported when Ceph Orchestrator is enabled.
+        If Ceph Orchestrator is not enabled, an empty list is returned.
+        :return: A list of all host labels.
+        """
+        labels = []
+        orch = OrchClient.instance()
+        if orch.available():
+            for host in orch.hosts.list():
+                labels.extend(host.labels)
+        return list(set(labels))  # Filter duplicate labels.
+
+    @handle_orchestrator_error('host')
+    def get(self, hostname: str) -> Dict:
+        """
+        Get the specified host.
+        :raises: cherrypy.HTTPError: If host not found.
+        """
+        return get_host(hostname)
+
+    @RESTController.Resource('PUT')
+    @raise_if_no_orchestrator
+    @handle_orchestrator_error('host')
+    def set_labels(self, hostname: str, labels: str):
```

Comment:
I'm not sure if we should introduce another two new endpoints. 
For getting labels, it's already in the response of `/api/host/<hostname>`.
For setting labels, can we just use `PUT /api/host/<hostname>`?

Reply:
I will introduce the endpoint `ui-api/host/labels` to get a list of all existing labels because this information is only consumed by the UI (auto-complete and select-badge fields).

Reply:
I changed the `set_labels` endpoint to `PUT /api/host/<hostname>`.